sys/include/net/gnrc/sixlowpan/nd/router.h

```diff
@@ -0,0 +1,164 @@
+#endif
+
+/**
+ * @brief   Representation for prefixes coming from a router
+ */
+typedef struct gnrc_sixlowpan_nd_router_prf {
+    struct gnrc_sixlowpan_nd_router_prf *next;  /**< next prefix */
+    gnrc_ipv6_netif_t *iface;                   /**< interface the prefix is registered too */
+    gnrc_ipv6_netif_addr_t *prefix;             /**< prefix on the interface/in the prefix list */
+} gnrc_sixlowpan_nd_router_prf_t;
+
+/**
+ * @brief   Abstract representation of a border router on all (border) routers.
+ */
+typedef struct {
+    ipv6_addr_t addr;                       /**< the IPv6 address of the border router (BR) */
+    uint32_t version;                       /**< version of the information dissiminated by the
+                                             *   BR */
+    uint16_t ltime;                         /**< the time in minutes until deletion */
+    BITFIELD(ctxs, GNRC_SIXLOWPAN_CTX_SIZE);/**< contexts associated with BR */
+    gnrc_sixlowpan_nd_router_prf_t *prfs;   /**< prefixes associated with BR */
+    vtimer_t ltimer;                        /**< timer for deletion */
+} gnrc_sixlowpan_nd_router_abr_t;
+
+/**
+ * @brief   Removes tentetative neighbor cache entries or sets registered ones to
+ *          garbage-collectible.
+ *
+ * @param[in] nc_entry  A neighbor cache entry.
+ */
+static inline void gnrc_sixlowpan_nd_router_gc_nc(gnrc_ipv6_nc_t *nc_entry)
+{
+    switch (gnrc_ipv6_nc_get_type(nc_entry)) {
+        case GNRC_IPV6_NC_TYPE_TENTATIVE:
+        case GNRC_IPV6_NC_TYPE_REGISTERED:
+            gnrc_ipv6_nc_remove(nc_entry->iface, &nc_entry->ipv6_addr);
+            break;
+        default:
+            break;
+    }
+
+}
+
+/**
+ * @brief   Set @p iface to router mode.
+ *
+ * @details This sets/unsets the GNRC_IPV6_NETIF_FLAGS_ROUTER and initializes or ceases router
+ *          behavior for 6LoWPAN neighbor discovery.
+ *
+ * @param[in] iface     An IPv6 interface. Must not be NULL.
+ * @param[in] enable    Status for the GNRC_IPV6_NETIF_FLAGS_ROUTER flag.
+ */
+static inline void gnrc_sixlowpan_nd_router_set_router(gnrc_ipv6_netif_t *netif, bool enable)
+{
+    if (enable) {
+        netif->flags |= GNRC_IPV6_NETIF_FLAGS_ROUTER;
+    }
+    else {
+        netif->flags &= ~GNRC_IPV6_NETIF_FLAGS_ROUTER;
+    }
+}
+
+/**
+ * @brief   Set/Unset GNRC_IPV6_NETIF_FLAGS_RTR_ADV flag for @p iface.
+ *
+ * @details GNRC_IPV6_NETIF_FLAGS_RTR_ADV and initializes or ceases
+ *          periodic router advertising behavior for neighbor discovery.
+ *
+ * @param[in] iface     An IPv6 interface. Must not be NULL.
+ * @param[in] enable    Status for the GNRC_IPV6_NETIF_FLAGS_RTR_ADV flag.
+ */
+static inline void gnrc_sixlowpan_nd_router_set_rtr_adv(gnrc_ipv6_netif_t *netif, bool enable)
+{
+    if (enable) {
+        netif->flags |= GNRC_IPV6_NETIF_FLAGS_RTR_ADV;
+    }
+    else {
+        netif->flags &= ~GNRC_IPV6_NETIF_FLAGS_RTR_ADV;
+    }
+}
```

Comment:
I don't think that anything gets more readable by replacing an assignment by a function and IMO it adds just 18 lines of additional code plus documentation, but this is not a blocker for this PR.